File: backend/app/src/main/java/com/zpi/app/entities/ParticipantTournament.java
package com.zpi.app.entities;

import javax.persistence.*;

@Entity
@Table(name ="tournaments_participants")
public class ParticipantTournament {

    @EmbeddedId
    private ParticipantTournamentID participantTournamentID;

    private String teamName;

    @ManyToOne
    @MapsId("participantId")
    private User participant;


    @ManyToOne
    @MapsId("tournamentId")
    private Tournament tournament;


    public ParticipantTournamentID getParticipantTournamentID() {
        return participantTournamentID;
    }

    public void setParticipantTournamentID(ParticipantTournamentID participantTournamentID) {
        this.participantTournamentID = participantTournamentID;
    }

    public String getTeamName() {
        return teamName;
    }

    public void setTeamName(String teamName) {
        this.teamName = teamName;
    }
}

File: frontend/tournaments-app/src/app/models/participantTournament.ts
import { User } from "./user";
import { Tournament } from "./tournament";
import { ParticipantTournamentID } from "./participantTournamentID";

export class ParticipantTournament {
  constructor(
    public participantTournamentID: ParticipantTournamentID,
    public teamName: String,
    public participant: User,
    public tournament: Tournament
  ) {}
}
